assets/js/ViewerApp.jsx

```diff
@@ -22,26 +24,38 @@ class ViewerApp extends Component {
       .join()
       .receive('ok', () => {});
 
-    channel.on('sign_update', ({ sign_id: signId, line_number: lineNumber, text: content }) => {
+    channel.on('sign_update', ({
+      sign_id: signId, duration, line_number: lineNumber, text: content,
+    }) => {
       this.setState(prevState => ({
-        signs: updateSigns(prevState.signs, { signId, lineNumber, content }),
+        signs: updateSigns(prevState.signs, {
+          signId, duration, lineNumber, content,
+        }),
       }));
     });
+
+    window.setInterval(this.updateTime, 1000);
```

Comment:
You should probably clear this interval with `clearInterval(id)` on unmount (on `componentWillUnmount`) see a similar example here: https://reactjs.org/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class

Reply:
cool, thanks!